k8s/pkg/registryserver/nse.go

```diff
@@ -52,11 +52,22 @@ func (rs *nseRegistryService) RegisterNSE(ctx context.Context, request *registry
 			logrus.Errorf("Failed to register nsm: %s", err)
 			return nil, err
 		}
-		nseResponse, err := rs.cache.AddNetworkServiceEndpoint(&v1.NetworkServiceEndpoint{
-			ObjectMeta: metav1.ObjectMeta{
+
+		var objectMeta metav1.ObjectMeta
+		if request.GetNetworkserviceEndpoint().EndpointName == "" {
+			objectMeta = metav1.ObjectMeta{
 				GenerateName: request.GetNetworkService().GetName(),
 				Labels:       labels,
-			},
+			}
+		} else {
+			objectMeta = metav1.ObjectMeta{
```

Comment:
In this case we need to check and probable update version of NSE, to be sure it is updated in case it is already exists.